### DISCUSSION TITLE
Query to match arbitrary-depth descendant?

### DISCUSSION THREAD

**Discussion by smackesey** [tree-sitter/tree-sitter#1817] [Q&A] [answered]
In shell globbing, I can write this to match all files named "bar" that live at arbitrary depth within "foo": `foo/**/bar`.

Is something similar possible in treesitter? The below pattern will only match if `bar` is a direct child of `foo`:

```
(foo
  (bar))
```

Is there any way to modify this query to make it match `bar` at _any_ depth inside foo?

**Comment by maxbrunsfeld** [comment, marked as answer]
This isn't possible right now, but I'd love to add some feature to make it possible at some point.

**Comment by alexmozaidze** [comment]
There is an open issue for this feature: #880

**Reply by amaanq** [reply]
Thanks, closing as duplicate
